### PR TITLE
Upgrade pyarrow minimum version to 7.0

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
Was this discussed somewhere?  I'm fine with bumping this pretty aggressively but not sure if we have a policy

**Comment by phofl** [comment]
Informal policy was support for packages released up to a year ago for 2.0, pyarrow 7 was released 2nd or 3rd of February 2022

**Comment by jreback** [comment]
+1 here for bumping

File: pandas/core/arrays/arrow/array.py [hunk 1]
```diff
@@ -567,7 +561,7 @@ def dropna(self: ArrowExtensionArrayT) -> ArrowExtensionArrayT:
         -------
         ArrowExtensionArray
         """
-        if pa_version_under6p0:
```
Comment (mroeschke) [inline pandas/core/arrays/arrow/array.py:570]: Can't this entire `if` statement be removed?

Comment (phofl) [reply pandas/core/arrays/arrow/array.py:570]: Oh must have missed that.

**Review by mroeschke — APPROVED** [review]

**Pull request merged**